configure.ac

```diff
@@ -2810,6 +2810,19 @@ AC_DEFINE_UNQUOTED(COROUTINE_H, ["$COROUTINE_H"])
 AC_SUBST(X_COROUTINE_H, [$COROUTINE_H])
 AC_SUBST(X_COROUTINE_SRC, [$COROUTINE_SRC])
 
+AC_CACHE_CHECK(for x86 control-flow protection, rb_cv_control_flow_protection,
+[AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[
+    void *shadow_stack_pointer = 0;
+    asm("rdsspq %0\n" : "=r" (shadow_stack_pointer));
```

Comment:
I'm not clear when this would pass or fail.

Would this test set `rb_cv_control_flow_protection=yes`...

1. If the CPU doesn't support shadow-stack? (I think not, the assembler mnemonic will be unknown, right?)
2. If the CPU supports shadow-stack, but the CFLAGS does not have `-fcf-protection` set? (I don't know)
3. If the CFLAGS have `-fcf-protection` set, but the CPU target doesn't support the feature? (I don't know)
4. If the CPU supports shadow-stack, and CFLAGS has `-fcf-protection`? (Presumably yes)